scrape/scrape.go

```diff
@@ -1005,6 +951,83 @@ mainLoop:
 	}
 }
 
+// scrapeAndReport is responsible for running a scape, and inserting with its
+// report in the storage, with a single appender.
```

Comment:
Suggested rewording: "scrapeAndReport performs a scrape and then appends the result to the storage together with reporting metrics, all using the same appender."